directory/cli/src/user.py

```diff
@@ -74,7 +74,93 @@ def show(login):
             error = '{}: user not found'.format(login)
             raise click.ClickException(error)
 
-    wrapper_commands = [
+    if not utils.advanced_mode_enabled():
+        @user.command(help='Create a new user')
+        def create():
+            click.echo('Please enter details for the following:')
+
+            params = {
+                        'login': click.prompt('  Username'),
+                        'first': click.prompt('  First name'),
+                        'last': click.prompt('  Surname'),
+                        'email': click.prompt('  Email'),
+            }
+            if click.confirm('Would you like to add a UID to this user?'):
+                params = { **params, 'uid': click.prompt('  UID') }
+
+            group_id = _get_group_id('clusterusers')
+
+            if group_id:
+                params = { **params, 'gidnumber': group_id[0] }
+
+            wrapper = ipa_wrapper_command.create_ipa_wrapper(
+                'user-add',
+                argument_name='login',
+                transform_options_callback=_transform_create_options,
+                handle_result_callback=_handle_create_result,
+            )
+            wrapper(**params)
+
+        @user.command(help='Modify an existing user')
+        def modify():
+            click.echo('Please enter the name of the user you want to modify:')
+            user = click.prompt('  Username')
+            user_find_args = ['--login={}'.format(user)]
+            try:
+                user_data = list_command.find_data(
+                        'user-find',
+                        user_find_args,
+                        all_fields=True
+                )[0]
+            except IpaRunError:
+                # Matches error shown for user show, can extract logic to a method
+                # in the future
+                error = '{}: user not found'.format(user)
+                raise click.ClickException(error)
+
+            click.echo('Modifying user: %s' % user)
+            params = {
+                'login': user,
+                'first': click.prompt(
+                    '  First name',
+                    default=user_data.get('First name', 'empty')[0]
```

Comment:
I'm not that fond of these lines, where we need to use `get` with a default rather than just indexing, since we're basically adding extra complexity and conditional handling just for the tests.

Firstly - why do we need the `[0]` - what does just the `.get` return normally? And won't this mean that when the default (`'empty'`) is used, we'll be indexing into this string, i.e. this will be like passing `prompt='e'` to `click.prompt`?

Ideally we would just stub `list_command.find_data` in the tests to give us the kind of result we'd get when really using IPA, and could then avoid that special handling. Failing that, could we use a `defaultdict` (similar to above) rather than repeatedly setting the same default, and could this default be something which would be likely to cause an obvious error if it ever occurred in practise (like an empty string)? Without doing that I can imagine these defaults actually being used in some future situation in practise, which would mean that a call to `user modify` and then hitting enter a few times could cause us to overwrite lots of a user's data with the string `'empty'`.

Reply:
Can these be removed now that we've improved the mocking in the tests?